hyperlink/__init__.py

```diff
@@ -1,8 +1,9 @@
 
-from ._url import URL, URLParseError, register_scheme, parse_host
+from ._url import URL, DecodedURL, URLParseError, register_scheme, parse_host
 
 __all__ = [
     "URL",
+    "DecodedURL",
```

Comment:
I want to say that we should just avoid exposing this entirely, but it probably needs to be exposed for type annotations.

However, as per #44, could we have a "decoded" property on `URL` that provides an interface to this, and an "encoded" property on `DecodedURL` that maps back to a `URL`?

(At this point I think I'm in favor of adding an `EncodedURL` alias for `URL`, then maybe adding a top-level entry point like `hyperlink.parse()` which takes a `decoded` kwarg flag which defaults to true, to make it easier to get started with `DecodedURL` which is what I think we all want most of the time.  That can definitely be deferred to a separate ticket though.)

Reply:
Yes, I am pretty much in favor of all those conveniences :) And I also agree that this probably needs to be exposed.